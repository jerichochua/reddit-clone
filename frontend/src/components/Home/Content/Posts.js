import React from 'react';
import { Link } from 'react-router-dom';
import Votes from '../../Vote/Votes';

const Posts = ({
  postId,
  title,
  author,
  score,
  timestamp,
  comments,
  type,
  url,
}) => {
  return (
    <li>
      <div className='post-container'>
        <Votes score={score} />
        <div className='post-item'>
          <div className='title'>
            {type === 'text' && <Link to={`/posts/${postId}`}>{title}</Link>}
            {type === 'link' && (
              <a href={url} target='_blank' rel='noreferrer'>
                {title}
              </a>
            )}
          </div>
          {type === 'link' && <div className='post-url'>{url}</div>}
          <div className='caption'>
            <span>by </span>
            <Link to={`/user/${author}`}>{author}</Link>
            <span> {timestamp}</span>
          </div>
          <div>
            <Link className='links' to={`/posts/${postId}`}>
              {comments} comment{comments === 1 ? '' : 's'}
            </Link>
          </div>
        </div>
      </div>
    </li>
  );
};

export default Posts;
