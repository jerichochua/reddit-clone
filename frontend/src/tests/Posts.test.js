import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Posts from '../components/Home/Content/Posts';

const post = {
  postId: '1',
  title: 'title',
  author: 'author',
  timestamp: 'timestamp',
  score: 1,
  comments: 1,
};

it('renders correctly', () => {
  const component = renderer.create(
    <MemoryRouter>
      <Posts {...post} />
    </MemoryRouter>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
