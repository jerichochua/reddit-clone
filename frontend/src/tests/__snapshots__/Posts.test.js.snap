// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<li>
  <div
    className="post-container"
  >
    <div
      className="votes-container"
    >
      <button
        className="upvote-btn"
        onClick={[Function]}
      />
      <span>
        1
      </span>
      <button
        className="downvote-btn"
        onClick={[Function]}
      />
    </div>
    <div
      className="post-item"
    >
      <div
        className="title"
      >
        <a
          href="/posts/1"
          onClick={[Function]}
        >
          title
        </a>
      </div>
      <div
        className="caption"
      >
        <span>
          by 
        </span>
        <a
          href="/user/author"
          onClick={[Function]}
        >
          author
        </a>
        <span>
           
          timestamp
        </span>
      </div>
      <div>
        <a
          className="links"
          href="/posts/1"
          onClick={[Function]}
        >
          1
           comment
        </a>
      </div>
    </div>
  </div>
</li>
`;
